app/src/main/java/com/duckduckgo/app/browser/defaultbrowsing/prompts/DefaultBrowserPromptsExperimentImpl.kt

```diff
@@ -265,20 +293,25 @@ class DefaultBrowserPromptsExperimentImpl @Inject constructor(
     }
 
     override fun onSetAsDefaultPopupMenuItemSelected() {
-        launchBestSelectionWindow()
+        fireInteractionPixel(AppPixelName.SET_AS_DEFAULT_IN_MENU_CLICK)
+        launchBestSelectionWindow(trigger = MENU)
     }
 
     override fun onMessageDialogShown() {
+        fireInteractionPixel(AppPixelName.SET_AS_DEFAULT_PROMPT_IMPRESSION)
     }
 
-    override fun onMessageDialogDismissed() {
+    override fun onMessageDialogCanceled() {
+        fireInteractionPixel(AppPixelName.SET_AS_DEFAULT_PROMPT_DISMISSED)
     }
 
     override fun onMessageDialogConfirmationButtonClicked() {
-        launchBestSelectionWindow()
+        fireInteractionPixel(AppPixelName.SET_AS_DEFAULT_PROMPT_CLICK)
+        launchBestSelectionWindow(trigger = DIALOG)
     }
 
     override fun onMessageDialogNotNowButtonClicked() {
+        fireInteractionPixel(AppPixelName.SET_AS_DEFAULT_PROMPT_DISMISSED)
```

Comment:
don’t we want to differentiate between dismissing a dialog and actively saying no?

Reply:
The pixel is defined as "triggered when user clicks 'Not now' or uses a gesture to close the prompt". In the end, regardless of whether user collapses the dialog or clicks not now, the intention is the same.